sdk/helper/pluginutil/run_config.go

```diff
@@ -106,6 +119,36 @@ func (rc runConfig) makeConfig(ctx context.Context) (*plugin.ClientConfig, error
 		},
 		AutoMTLS: rc.AutoMTLS,
 	}
+	if rc.image == "" {
+		clientConfig.Cmd = cmd
+		clientConfig.SecureConfig = &plugin.SecureConfig{
+			Checksum: rc.sha256,
+			Hash:     sha256.New(),
+		}
+	} else {
+		clientConfig.SkipHostEnv = true
+		clientConfig.RunnerFunc = func(logger hclog.Logger, goPluginCmd *exec.Cmd, tmpDir string) (runner.Runner, error) {
+			overlayCmdSpec(goPluginCmd, cmd)
+			cfg := &config.ContainerConfig{
+				UnixSocketGroup: fmt.Sprintf("%d", os.Getgid()),
+				Image:           rc.image,
+				Tag:             rc.imageTag,
+				SHA256:          fmt.Sprintf("%x", rc.sha256),
+				Labels: map[string]string{
+					"managed-by": "hashicorp.com/vault",
+				},
+				// TODO: More configurables.
+				// Defaulting to runsc will require installing gVisor in the GitHub runner.
```

Comment:
We will probably want to check that the `runsc` binary is available and executable before defaulting to it, too?

Reply:
I forgot to note my plans here in the description, but that's a good shout. What I'll probably do is error if `runsc` isn't available in $PATH, and if people want to use `runc` for example, they'll need to create their own Vault "plugin runtime" in the Vault API and reference it when registering the plugin.

That lets us use a safe default that behaves predictably, and hopefully still with a helpful enough UX, because I wouldn't want Vault to suddenly start using a different container runtime if `runsc` gets uninstalled out of band.